src/stats.rs

```diff
@@ -2,8 +2,52 @@
 
 use crate::definitions::Image;
 use image::{GenericImageView, GrayImage, Pixel, Primitive};
+use itertools::Itertools;
 use num::Bounded;
 
+/// A minimum and maximum value returned by [`minmax()`]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct MinMax<T> {
+    /// The minimum value
+    pub min: T,
+    /// The maximum value
+    pub max: T,
+}
+
+/// Returns the minimum and maximum values per channel in an image.
+pub fn minmax<P, T>(image: &Image<P>) -> Vec<MinMax<T>>
```

Comment:
```suggestion
pub fn min_max<P, T>(image: &Image<P>) -> Vec<MinMax<T>>
```

Reply:
My thoughts on this is that `minmax` is more consistent with `Itertools::minmax` as well as being the same name as the `MinMax` struct. It doesn't really matter which one to be fair though.

Reply:
Let’s go with `min_max`; the Itertools name seems odd here.